packages/fluentui/react-northstar/src/components/Attachment/Attachment.tsx

```diff
@@ -49,103 +53,130 @@ export interface AttachmentProps extends UIComponentProps, ChildrenComponentProp
-    icon: customPropTypes.shorthandAllowingChildren,
-    progress: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
-  };
-
-  static defaultProps = {
-    accessibility: attachmentBehavior as Accessibility,
-  };
-
-  renderComponent({ ElementType, classes, unhandledProps, styles, variables, accessibility }) {
-    const { header, description, icon, action, progress } = this.props;
-
-    return (
-      <ElementType
-        className={classes.root}
-        onClick={this.handleClick}
-        {...accessibility.attributes.root}
-        {...unhandledProps}
-        {...applyAccessibilityKeyHandlers(accessibility.keyHandlers.root, unhandledProps)}
-      >
-        {icon &&
-          Box.create(icon, {
-            defaultProps: () => ({ styles: styles.icon }),
-          })}
-        {(header || description) && (
-          <div className={classes.content}>
-            {Text.create(header, {
-              defaultProps: () => ({ styles: styles.header }),
-            })}
-
-            {Text.create(description, {
-              defaultProps: () => ({ styles: styles.description }),
-            })}
-          </div>
-        )}
-        {action &&
-          Button.create(action, {
-            defaultProps: () => ({
-              iconOnly: true,
-              text: true,
-              styles: styles.action,
-              className: Attachment.slotClassNames.action,
-            }),
-          })}
-        {!_.isNil(progress) && <div className={classes.progress} />}
-      </ElementType>
-    );
-  }
-
-  actionHandlers = {
-    performClick: event => this.performClick(event),
-  };
-
-  performClick = e => {
-    if (e.currentTarget === e.target) {
-      e.stopPropagation();
-      this.handleClick(e);
-    }
-  };
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables,
+    }),
+    rtl: context.rtl,
+  });
 
-  handleClick = (e: React.SyntheticEvent) => {
-    const { disabled } = this.props;
+  const ElementType = getElementType(props);
+  const unhandledProps = useUnhandledProps(Attachment.handledProps, props);
 
+  const handleClick = (e: React.KeyboardEvent | React.MouseEvent) => {
     if (disabled) {
       e.preventDefault();
       return;
     }
 
-    _.invoke(this.props, 'onClick', e, this.props);
+    _.invoke(props, 'onClick', e, props);
   };
-}
+
+  const element = (
+    <ElementType {...getA11Props('root', { className: classes.root, onClick: handleClick, ...unhandledProps })}>
+      {Box.create(icon, {
+        defaultProps: () => ({ styles: resolvedStyles.icon }),
+      })}
+      {(header || description) && (
+        <div className={classes.content}>
+          {Text.create(header, {
+            defaultProps: () => ({ styles: resolvedStyles.header }),
+          })}
+
+          {Text.create(description, {
+            defaultProps: () => ({ styles: resolvedStyles.description }),
+          })}
+        </div>
+      )}
+      {Button.create(action, {
```

Comment:
Please why don't you have check that `action` is defined, compared with the previous code?

Reply:
Because it will be done by `.create()` factory 😺